server/src/LspServer.ts

```diff
@@ -149,25 +136,78 @@ export class LspServer {
+    if (this.contextFirstInitPromiseResolve) {
+      this.contextFirstInitPromiseResolve(true);
+    }
+
+    if (bigQueryContextInfo.isOk()) {
+      this.bigQueryContext = bigQueryContextInfo.value;
+      return this.bigQueryContext.contextInfo;
+    }
+
+    return this.showErrorDialog<ContextInfo>(
+      `Only common dbt features will be available. Dbt profile was not configured. ${bigQueryContextInfo.error.error}`,
+      'warning',
+      this.prepareDestination.bind(this),
+      () => Promise.resolve(bigQueryContextInfo.error),
+    );
+  }
+
+  async prepareRpcServer(): Promise<void> {
+    this.initDbtRpcAttempt++;
+
     const [command, dbtPort] = await Promise.all([
       this.featureFinder.findDbtRpcCommand(this.connection.sendRequest('custom/getPython')),
       this.featureFinder.findFreePort(),
     ]);
 
     if (command === undefined) {
-      const errorMessageResult = await this.connection.window.showErrorMessage(
+      this.featureFinder = new FeatureFinder();
+      return this.showErrorDialog(
         `Failed to find dbt-rpc. You can use 'python3 -m pip install dbt-bigquery dbt-rpc' command to install it. Check in Terminal that dbt-rpc works running 'dbt-rpc --version' command or [specify the Python environment](https://code.visualstudio.com/docs/python/environments#_manually-specify-an-interpreter) for VS Code that was used to install dbt (e.g. ~/dbt-env/bin/python3).`,
-        { title: 'Retry', id: 'retry' },
+        'error',
+        this.prepareRpcServer.bind(this),
+        () => Promise.resolve(),
       );
-      if (errorMessageResult?.id === 'retry') {
-        this.featureFinder = new FeatureFinder();
-        await this.onInitialized();
-      }
-      return;
     }
 
     command.addParameter(dbtPort.toString());
-    await Promise.all([this.startDbtRpc(command, dbtPort), this.zetaSqlWrapper.initializeZetaSql()]);
+    return this.startDbtRpc(command, dbtPort);
+  }
+
+  async showErrorDialog<T>(message: string, messageType: 'warning' | 'error', retry: () => Promise<T>, discontinue: () => Promise<T>): Promise<T> {
+    const actions = { title: 'Retry', id: 'retry' };
+    const errorMessageResult =
+      messageType === 'warning'
+        ? await this.connection.window.showWarningMessage(message, actions)
+        : await this.connection.window.showErrorMessage(message, actions);
+    if (errorMessageResult?.id === 'retry') {
+      return retry();
+    }
+    return discontinue();
+  }
+
+  logStartupInfo(contextInfo: ContextInfo, initTime: number, initDestinationAttempt: number, initDbtRpcAttempt: number): void {
+    this.sendTelemetry('log', {
+      dbtVersion: getStringVersion(this.featureFinder.version),
+      python: this.featureFinder.python ?? 'undefined',
+      initTime: initTime.toString(),
+      type: contextInfo.type ?? 'unknown type',
+      method: contextInfo.method ?? 'unknown method',
+      initDestinationAttempt: initDestinationAttempt.toString(),
```

Comment:
In which case do we need to retry destination initialization?

Reply:
It make sense for oauth BQ authentication method. This ability removed for now.